ui/pages/confirmations/components/confirm/info/info.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { memo, useMemo } from 'react';
+import { useSelector } from 'react-redux';
+
+import { MESSAGE_TYPE } from '../../../../../../shared/constants/app';
+import {
+  BackgroundColor,
+  BorderRadius,
+} from '../../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+import { currentConfirmationSelector } from '../../../../../selectors';
+import { Box } from '../../../../../components/component-library';
+import {
+  ConfirmInfo,
+  ConfirmInfoRowType,
+} from '../../../../../components/app/confirm/info/info';
+
+const Info: React.FC = memo(() => {
+  const t = useI18nContext();
+  const currentConfirmation = useSelector(currentConfirmationSelector);
+
+  const infoRows = useMemo(() => {
+    if (
+      !currentConfirmation ||
+      currentConfirmation.type !== MESSAGE_TYPE.PERSONAL_SIGN ||
+      !currentConfirmation.msgParams?.origin
+    ) {
+      return undefined;
+    }
+    return [
+      {
+        label: t('origin'),
+        type: ConfirmInfoRowType.UrlType,
+        rowProps: {
+          url: currentConfirmation.msgParams?.origin,
+        },
+      },
+    ];
+  }, [currentConfirmation]);
```

Comment:
currently, I don't think useMemo is making much of an improvement here if any, though the computation here will grow in size as we support more use cases. Since useMemo doesn't natively support deep comparisons, this may become stale with updates to the confirmation props.

maybe we could remove useMemo here and consider re-adding it with a deep comparison later. thoughts?

Reply:
I do not expect changes in current confirmation @digiwand

Reply:
makes sense. it would be more of a preventative measure at this point in the pursuit of creating robust code